src/SegmentationCameraSensor.cc

```diff
@@ -183,24 +169,50 @@ bool SegmentationCameraSensor::Load(const sdf::Sensor &_sdf)
 
   this->dataPtr->sdfSensor = _sdf;
 
-  // Create the thermal image publisher
-  this->dataPtr->publisher =
+  // Segmentation type
+  sdf::ElementPtr sdfElement = _sdf.Element();
+  if (sdfElement->HasElement("segmentation_type"))
+  {
+    auto type = sdfElement->Get<std::string>("segmentation_type");
+
+    // convert type to lowercase
+    std::for_each(type.begin(), type.end(), [](char & c){
+      c = std::tolower(c);
+    });
+
+    if (type == "semantic")
+      this->dataPtr->type = rendering::SegmentationType::Semantic;
+    else if (type == "instance" || type == "panoptic")
+      this->dataPtr->type = rendering::SegmentationType::Panoptic;
+  }
+
+  // Create the segmentation colored map image publisher
+  this->dataPtr->coloredMapPublisher =
+      this->dataPtr->node.Advertise<ignition::msgs::Image>(
+          this->Topic() + this->dataPtr->topicColoredMap);
+
+  // Create the segmentation labels map image publisher
+  this->dataPtr->labelsMapPublisher =
       this->dataPtr->node.Advertise<ignition::msgs::Image>(
-          this->Topic());
+          this->Topic() + this->dataPtr->topicLabelsMap);
 
-  if (!this->dataPtr->publisher)
+  if (!this->dataPtr->labelsMapPublisher || !this->dataPtr->coloredMapPublisher)
   {
     ignerr << "Unable to create publisher on topic["
       << this->Topic() << "].\n";
     return false;
   }
 
-  if (!this->AdvertiseInfo())
+  if (!this->AdvertiseInfo(this->Topic() + "/camera_info"))
```

Comment:
In all similar sensors implementations, they just call  `this->AdvertiseInfo()` directly .. but it publishes on topic "/camera_info" .. which may conflict with other camera ... so i add its name to the topic first 

I cannot access the "/camera_info" string other than hardcoding it .. because it is hardcoded in the parent camera class

Reply:
> I cannot access the "/camera_info" string other than hardcoding it .. because it is hardcoded in the parent camera class

:+1: 

It looks like there's a `TODO` note about making the camera info topic configurable in the parent camera class:

https://github.com/ignitionrobotics/ign-sensors/blob/53c87ea5be1796fa4c3c5fce376a07c8bd695101/src/CameraSensor.cc#L491-L492

So, would you mind adding a `TODO` note here as well to update the hardcoded topic once the parent camera class is changed to make this topic configurable? That way, we keep this class up-to-date whenever the parent camera class is updated.